apps/api/src/donations/donations.service.ts

```diff
@@ -114,6 +115,87 @@ export class DonationsService {
     return donation
   }
 
+  /**
+   * Create initial donation object for tracking purposes
+   * This is used when the payment is created from the payment intent
+   */
+  async createInitialDonationFromIntent(
+    campaign: Campaign,
+    stripePaymentDto: CreateStripePaymentDto,
+    paymentIntent: Stripe.PaymentIntent,
+  ): Promise<Donation> {
+    Logger.debug('[ CreateInitialDonationFromIntent]', {
+      campaignId: campaign.id,
+      amount: paymentIntent.amount,
+      paymentIntentId: paymentIntent.id,
+    })
+
+    /**
+     * Create or connect campaign vault
+     */
+    const vault = await this.prisma.vault.findFirst({ where: { campaignId: campaign.id } })
+    const targetVaultData = vault
+      ? // Connect the existing vault to this donation
+        { connect: { id: vault.id } }
+      : // Create new vault for the campaign
+        { create: { campaignId: campaign.id, currency: campaign.currency, name: campaign.title } }
+
+    /**
+     * Create or update initial donation object
+     */
+    const donation = await this.prisma.donation.upsert({
+      where: { extPaymentIntentId: paymentIntent.id },
+      create: {
+        amount: 0, //this will be updated on successful payment event
+        chargedAmount: paymentIntent.amount,
+        currency: campaign.currency,
+        provider: PaymentProvider.stripe,
+        type: DonationType.donation,
+        status: DonationStatus.initial,
+        extCustomerId: stripePaymentDto.personEmail ?? '',
```

Comment:
This looks like a data inconsistency - to store an email in the ID field.

Normally I think the values stored in this column should start with `cus_...`

Reply:
We should probably change it in the current implementation too then, because it is the same https://github.com/podkrepi-bg/api/pull/425/files/ba5b5f56e1d5c68b51284575b93fd9c0f06a9882#diff-3941d803ef3b8b94eac319c8965e8ea1a254e002f4f3878dae61f4b3ca783c15R82

Reply:
I was thinking that this email is what connects the donation to the user

Reply:
Looks like this field is a mess...

https://github.com/podkrepi-bg/api/blob/9d5e7a80c0d3ca2a4add449b9d7dc1b01fbb3b58/apps/api/src/bank-transactions-file/helpers/parser.ts#L27


https://github.com/podkrepi-bg/api/blob/54273881695cc1290ae9c3ba370f676ea8148024/apps/api/src/campaign/campaign.service.ts#L468

https://github.com/podkrepi-bg/api/blob/a89b7007edbc46c0151edad6c3095315a0077b27/apps/api/src/donations/events/stripe-payment.service.ts#L168

https://github.com/podkrepi-bg/api/blob/5b0589f3e162eee0034ef8a3acdf1b450339168b/apps/api/src/donations/donations.service.ts#L81

I'd say let's leave it as is and create a new issue to improve the situation and the data integrity.

Reply:
https://github.com/podkrepi-bg/api/issues/428